<a href="https://colab.research.google.com/github/linyu2703/nhanes_inferential_2021_23/blob/main/inferential.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [15]:
# importing libraries
import pandas as pd
import numpy as np
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind
import statsmodels.api as sm
from statsmodels.formula.api import ols
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import matplotlib.pyplot as plt

In [16]:
# loading and reading datasets
demo_src = pd.read_sas('DEMO_L.xpt')
bpxo_src = pd.read_sas('BPXO_L.xpt')
hepb_src = pd.read_sas('HEPB_S_L.xpt')
vid_src = pd.read_sas('VID_L.xpt')
kiq_src = pd.read_sas('KIQ_U_L.xpt')
paq_src = pd.read_sas('PAQ_L.xpt')
whq_src = pd.read_sas('WHQ_L.xpt')

In [17]:
# creating a separate dataframe from the main src dataframe with specific columns
# creating a main dataframe with those dataframes merged by SEQN along with the other specified columns
demo = demo_src[['SEQN','DMDMARTZ','DMDEDUC2','RIDAGEYR']].copy()
bpxo = bpxo_src[['SEQN','BPXOSY3','BPXODI3']].copy()
hepb = hepb_src[['SEQN', 'LBXHBS']].copy()
vid = vid_src[['SEQN', 'LBDVD2LC']].copy()
kiq = kiq_src[['SEQN', 'KIQ022']].copy()
paq = paq_src[['SEQN', 'PAD680']].copy()
whq = whq_src[['SEQN', 'WHD020']].copy()
df = demo.copy()
df = (demo
      .merge(bpxo, on='SEQN', how='left')
      .merge(hepb, on='SEQN', how='left')
      .merge(vid, on='SEQN', how='left')
      .merge(kiq, on='SEQN', how='left')
      .merge(paq, on='SEQN', how='left')
      .merge(whq, on='SEQN', how='left')
      )

In [19]:
# renaming columns and recoding values for clarity
df_cleaned = df[['SEQN','DMDMARTZ','DMDEDUC2','RIDAGEYR','BPXOSY3','BPXODI3','LBXHBS','LBDVD2LC','KIQ022','PAD680','WHD020']].copy().rename(columns={
    'SEQN': 'ID',
    'DMDMARTZ': 'Marital Status',
    'DMDEDUC2': 'Education Level',
    'RIDAGEYR': 'Age',
    'BPXOSY3': 'Systolic BP',
    'BPXODI3': 'Diastolic BP',
    'LBXHBS': 'Hep B',
    'LBDVD2LC': 'Vitamin D',
    'KIQ022': 'Kidney Condition',
    'PAD680': 'Physical Activity',
    'WHD020': 'Weight'
})

df_cleaned['Marital Status'] = np.where(df_cleaned['Marital Status'] == 1, 'married',
                                   np.where(df_cleaned['Marital Status'].isin([2,3,77,99]), 'not married', pd.NA))
df_cleaned['Education Level'] = np.where(df_cleaned['Education Level'].isin([4,5]),'bachelors or higher',
                                    np.where(df_cleaned['Education Level'].isin([1,2,3,7,9]),'less than bachelors', pd.NA))
df_cleaned['Age'] = df_cleaned['Age'].mask(df_cleaned['Age'].abs() < 1e-10, np.nan)
df_cleaned['Kidney Condition'] = np.where(df_cleaned['Kidney Condition'] == 1, 'Yes', np.where(df_cleaned['Kidney Condition'] == 2, 'No', pd.NA))
df_cleaned['Physical Activity'] = df_cleaned['Physical Activity'].replace([7777,9999], np.nan)
df_cleaned['Weight'] = df_cleaned['Weight'].replace([7777,9999], np.nan)

df_cleaned['Physical Activity'] = df_cleaned['Physical Activity'].fillna(round(df_cleaned['Physical Activity'].median(), 0))
df_cleaned['Weight'] = df_cleaned['Weight'].fillna(round(df_cleaned['Weight'].mean(), 0))

In [22]:
# Question 1: "Is there an association between marital status (married or not married) and education level (bachelor’s degree or higher vs. less than a bachelor’s degree)?"
# chi-squared test will be used to test whether the two variables: (martial status and education level) are independent or dependent
Q1 = df_cleaned.loc[(df_cleaned['Marital Status'].isin(['married','not married'])) &
    (df_cleaned['Education Level'].isin(['bachelors or higher','less than bachelors']))
].copy()

Q1['Marital Status'].value_counts()
Q1['Education Level'].value_counts()

ct = pd.crosstab(Q1['Marital Status'], Q1['Education Level'])
print(ct)

# Performing chi-square analysis
chi2, p, _, _ = chi2_contingency(ct)
print("Chi-square =", round(chi2, 2))
print("p value =", p)


Education Level  bachelors or higher  less than bachelors
Marital Status                                           
married                         2782                 1354
not married                     2212                 1444
Chi-square = 38.24
p value = 6.264475724779772e-10


p value is less than 0.05, which means there is a significant correlation between martial status and education level. people with bachelors degree or higher are more likely to be married compared to those with less than bachelors degree.

In [24]:
# Question 2: "Is there a difference in the mean sedentary behavior time between those who are married and those who are not married?"
# 2 sample t-test will be performed to test the relationship between maritial status and physical actvity
# null hypothesis: no difference in mean sedentary time between married and not married people
# alternative hypothesis: there is a difference in mean sedentary time between married and not married people
Q2 = df_cleaned.loc[
    (df_cleaned['Marital Status'].isin(['married','not married'])) &
    (df_cleaned['Physical Activity'] != 'NaN')
].copy()

married = Q2.loc[Q2['Marital Status']=='married', 'Physical Activity']
not_married = Q2.loc[Q2['Marital Status']=='not married', 'Physical Activity']

t_stat, p_val = ttest_ind(married, not_married, equal_var=False)
print("t-statistic =", round(t_stat, 2))
print("p value =", p_val )

married_mean = married.mean()
not_married_mean = not_married.mean()
print("Married Mean =", round(married_mean, 2))
print("Not Married Mean =", round(not_married_mean, 2))

t-statistic = -3.78
p value = 0.00015846852964065343
Married Mean = 352.9
Not Married Mean = 371.03


p value is less than 0.05 which means there is a correlation between martial status and physical activity, the null hypothesis is rejected. the t-stat is -3.78 meaning married people have a lowered mean than not married people, so married people are more likely to be less sedentary compared to than those who are not married.